### PR TITLE
Unique mlir kernel names

### PR DESCRIPTION
**Pull request by kushanam**
This MR incorporates the name attribute for kernel get in MLIR. This is done to enable generation of the experimental kernels for int64 bit indexing.

CC: @frgossen

### CODE REVIEW

File: tensorflow/core/kernels/mlir_generated/BUILD [hunk 1]
```diff
@@ -796,18 +796,18 @@ gpu_kernel_library(
...
-#     tile_size = "256",
-#     types = [
-#         "f32",
-#         "f64",
-#     ],
-#     unroll_factors = "4",
-#     jit_i64_indexed_for_large_tensors_types = ["f16"],
-# )
+#TODO: enable when supported in the pipeline
+gpu_kernel_library(
```
Comment (frgossen) [inline tensorflow/core/kernels/mlir_generated/BUILD:800]: Are you sure, enabeling this already passes the tests?

**Review by frgossen — CHANGES_REQUESTED** [review]

File: tensorflow/core/kernels/mlir_generated/BUILD (hunk 1 above)
Comment (frgossen) [reply tensorflow/core/kernels/mlir_generated/BUILD:800]: This fails the internal tests :/

Comment (kushanam) [reply tensorflow/core/kernels/mlir_generated/BUILD]: This test is disabled for now per our discussion.

**Review by frgossen — APPROVED** [review]
Thanks!

**Pull request merged**